### PR TITLE
feat(rules): split button name rule into button only and button input rules

### PR DESCRIPTION
**Pull request by scurker**
Split `button-name` into `button-name` and `input-button-name` rules.

* `button-name` rule should only include `button` and `[role=button]`
* `input-button-name` rule should only include inputs where type is `button`, `reset`, or `submit`.

For each rule, removed any checks that were no longer applicable to the targeted elements.

Closes: #1326 

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**
- [x] Follows the commit message policy, appropriate for next version
- [x] Has documentation updated, a DU ticket, or requires no documentation change
- [x] Includes new tests, or was unnecessary
- [x] Code is reviewed for security by: @WilcoFiers

### CODE REVIEW

**Comment by straker** [comment]
LGTM but will need @WilcoFiers approval before we merge

File: lib/rules/input-button-name.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+	"id": "input-button-name",
```
Comment (stephenmathieson) [inline lib/rules/input-button-name.json]: Funky spacing. Did Prettier not run here?

Comment (scurker) [reply lib/rules/input-button-name.json]: How is the spacing funky? It looks normal to me.

https://github.com/dequelabs/axe-core/blob/a2c531dd9bc812a639f742cc9b156d2869748b8a/lib/rules/input-button-name.json#L1-L10

Comment (stephenmathieson) [reply lib/rules/input-button-name.json]: Weird. Must have been a Github Refined bug. It looked like indentation was set to 1 space. :man_shrugging:

Comment (scurker) [reply lib/rules/input-button-name.json]: Huh, I don't think it is. I think something funky is going on with Github's formatting. There's different spacing from when I looked at it earlier. 👀

File: lib/rules/input-button-name.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	"any": [
+		"non-empty-if-present",
+		"non-empty-value",
+		"aria-label",
+		"aria-labelledby",
+		"role-presentation",
+		"role-none",
+		"non-empty-title"
+	],
+	"none": ["focusable-no-name"]
```
[the remark this answers is not in the document]
Comment (scurker) [reply lib/rules/input-button-name.json]: Looks like it's not still necessary, I removed it. Thanks!

**Pull request merged**